packages/context/src/inject.ts

```diff
@@ -282,6 +319,12 @@ function resolveAsGetter(
   injection: Readonly<Injection>,
   session?: ResolutionSession,
 ) {
+  const targetType = inspectTargetType(injection);
+  if (targetType && targetType !== Function) {
+    throw new Error(
+      `The target type ${targetType.name} is not a Getter function`,
```

Comment:
+1 to detect this common case of programmer errors.

I am concerned about the overall UX though.

When the user receives this error, how can they find the place that has triggered it? Is the error stack going to contain the file name + line number of the place where `@inject.getter` was called incorrectly? Is there any mechanism adding binding keys and/or names of the decorated classes to the error message, so that developers at least know what binding keys to search for?

Reply:
> +1 to detect this common case of programmer errors.
>
> I am concerned about the overall UX though.
>
> When the user receives this error, how can they find the place that has triggered it? Is the error stack going to contain the file name + line number of the place where @inject.getter was called incorrectly? Is there any mechanism adding binding keys and/or names of the decorated classes to the error message, so that developers at least know what binding keys to search for?

PTAL